### DISCUSSION TITLE
Presenting the Git commit ID in the driver during runtime/debug time

### DISCUSSION THREAD

**Discussion by sskras** [openzfsonwindows/openzfs#407] [Ideas] [answered]
After reading https://github.com/openzfsonwindows/openzfs/issues/401#issuecomment-2423700425 I thought:

would it be worth storing the commit hash used to build the driver somewhere so it can later be revealed in `info.txt`, `cbuf.txt`, or `stack.txt`, or probably all of them?

**Comment by lundman** [comment, marked as answer]
Technically we do have the build number, its in zfs_gitrev.h and used in Registry, `zpool version` etc. But `cbuf` is a circular buffer so as it is printed first it is lost. But one could add to spl-panic() to print the version as the last line - which could be useful.
In debugger, you can just lookup `zfs_version`

**Reply by sskras** [reply]
Thanks, will do!

**Reply by sskras** [reply]
PS. Printing it during the panic surely seems like a nice addition to have.

I guess that's the source:
https://github.com/openzfsonwindows/openzfs/blob/740220fab77fef102a62a08ce642a2c17025f660/include/zfs_gitrev.h.win

**Reply by lundman** [reply]
It uses that file to generate `zfs_gitrev.h` from cmake. spl-debug.c I believe is where panic() is defined too, I added it to print the panic string to cbuf there recently.

**Comment by sskras** [comment]
Yay! @lundman fixed this right away: 
https://github.com/openzfsonwindows/openzfs/commit/f287afab7dc0ee24dc210852971c6b0685685a2c#diff-7560b396bab0bd6f57402db4d2f6252c7f47f6dbdc4e5e4957e5dfbaebbcded3R82

Thank you :)
